synapse/tools/promote.py

```diff
@@ -29,7 +32,18 @@ async def main(argv, outp=s_output.stdout):
             graceful = not opts.failure
 
             outp.printf(f'Promoting to leader: {opts.svcurl}')
-            await cell.promote(graceful=graceful)
+            try:
+                await cell.promote(graceful=graceful)
+            except s_exc.BadState as e:
+                mesg = f'Failed to promote service {s_urlhelp.sanitizeUrl(opts.svcurl)} to being a leader; it is' \
```

Comment:
What about including the exception message in this message instead?  I'm thinking that this assumes BadState will only ever be raised for the case in this mesg, which we would have to remember in the future if we raise that elsewhere in the promotion process.

Reply:
Good catch. We can probably just print Error: BadState {mesg} or something :+1: